tests/Unit/Utilities/CssDocumentTest.php

```diff
@@ -0,0 +1,403 @@
+            'with line feeds within rule' => ['screen', "\n"],
+            'with Windows line endings within rule' => ['screen', "\r\n"],
+            'with TABs within rule' => ['screen', "\t"],
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * @param string $mediaQuery
+     *
+     * @dataProvider provideMediaQueryWithTvType
+     */
+    public function discardsMediaRuleWithTypeNotInAllowlist(string $mediaQuery): void
+    {
+        $subject = new CssDocument('@media ' . $mediaQuery . ' { p { color: red; } }');
+
+        $result = $subject->getStyleRulesData(['screen']);
+
+        self::assertCount(0, $result);
+    }
+
+    /**
+     * @return array<string, array<int, string>>
+     */
+    public function provideMediaQueryWithTvType(): array
+    {
+        return [
+            'type alone' => ['tv'],
+            'type with `only`' => ['only tv'],
+            'type and plain value' => ['tv and (max-width: 480px)'],
+            'type and plain value with `not`' => ['tv and not (max-width: 480px)'],
+            'type and plain values with `and`' => ['tv and (min-width: 320px) and (max-width: 480px)'],
+            'type and range (singly bounded)' => ['tv and (height > 600px)'],
+            'type and range (fully bounded)' => ['tv and (400px <= width <= 700px)'],
+            'with line feed before' => ["\ntv"],
+            'with Windows line ending before' => ["\r\ntv"],
+            'with TAB before' => ["\ttv"],
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * @param string $cssBetween
+     * @param string $cssBefore
+     *
+     * @dataProvider provideCssWithoutStyleRules
+     */
+    public function parsesMultipleStyleRules(string $cssBetween, string $cssBefore = ''): void
+    {
+        $subject = new CssDocument(
+            $cssBefore . 'p { color: green; }' . $cssBetween . '@media screen { h1 { color: green; } }'
+        );
+
+        $result = $subject->getStyleRulesData(['screen']);
+
+        // The content of the parsed rules is covered by other tests.  Here just check the number of parsed rules.
+        self::assertCount(2, $result);
+    }
+
+    /**
+     * @return array<string, array<int, string>>
+     */
+    public function provideCssWithoutStyleRules(): array
+    {
+        return [
+            'nothing' => [''],
+            'space' => [' '],
+            'line feed' => ["\n"],
+            'Windows line ending' => ["\r\n"],
+            'TAB' => ["\r\n"],
+            'non-conditional at-rule (valid `@font-face`)' => [self::VALID_AT_FONT_FACE_RULE],
+            'comment' => ['/* Test */'],
+            'commented-out style rule' => ['/* p { color: red; } */'],
+            'space before' => ['', ' '],
+            'line feed before' => ['', "\n"],
+            'Windows line ending before' => ['', "\r\n"],
+            'TAB before' => ['', "\r\n"],
+            'non-conditional at-rule (`@charset`) before' => ['', '@charset "UTF-8";'],
```

Comment:
Please let's split this up as well.

Reply:
Splitting this one up works well, as the same data provider can be used for both tests.